From 4e54e2f4e0bba53d6ddbd440e308bd9e31885905 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Tue, 29 Jan 2019 14:51:22 +0500
Subject: [PATCH 1/3] udev-extraconf/mount.sh: define mount prefix using a
 variable

This allows easy manipulation of the prefix when and where
needed so the hardcoded string could be managed from a single
location.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 mount.sh | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

Index: 1.1-r0/mount.sh
===================================================================
--- 1.1-r0.orig/mount.sh
+++ 1.1-r0/mount.sh
@@ -25,6 +25,8 @@ else
 fi
 
 PMOUNT="/usr/bin/pmount"
+UMOUNT="/bin/umount"
+MOUNT_PREFIX="/run/media"
 
 for line in `grep -h -v ^# /etc/udev/mount.blacklist /etc/udev/mount.blacklist.d/*`
 do
@@ -95,7 +97,7 @@ automount() {
 		name="${LABEL}-${name}"
 	fi
 
-	! test -d "/run/media/$name" && mkdir -p "/run/media/$name"
+	! test -d "$MOUNT_PREFIX/$name" && mkdir -p "$MOUNT_PREFIX/$name"
 	# Silent util-linux's version of mounting auto
 	if [ "x`readlink $MOUNT`" = "x/bin/mount.util-linux" ] ;
 	then
@@ -113,12 +115,12 @@ automount() {
 		;;
 	esac
 
-	if ! $MOUNT -t auto $DEVNAME "/run/media/$name"
+	if ! $MOUNT -t auto $DEVNAME "$MOUNT_PREFIX/$name"
 	then
-		#logger "mount.sh/automount" "$MOUNT -t auto $DEVNAME \"/run/media/$name\" failed!"
-		rm_dir "/run/media/$name"
+		#logger "mount.sh/automount" "$MOUNT -t auto $DEVNAME \"$MOUNT_PREFIX/$name\" failed!"
+		rm_dir "$MOUNT_PREFIX/$name"
 	else
-		logger "mount.sh/automount" "Auto-mount of [/run/media/$name] successful"
+		logger "mount.sh/automount" "Auto-mount of [$MOUNT_PREFIX/$name] successful"
 		touch "/tmp/.automount-$name"
 	fi
 }
@@ -157,5 +159,5 @@ if [ "$ACTION" = "remove" ] || [ "$ACTIO
 
     # Remove empty directories from auto-mounter
     name="`basename "$DEVNAME"`"
-    test -e "/tmp/.automount-$name" && rm_dir "/run/media/$name"
+    test -e "/tmp/.automount-$name" && rm_dir "$MOUNT_PREFIX/$name"
 fi
